### PR TITLE
net-ftp/pure-ftpd: migrate to stand-alone configuration

### PR DESCRIPTION
**Pull request by ConiKost**
This bump changed at lot of things:
* Deprecating configuration via 'conf.d', since not all option are
  available via command line. User will be informed about this.
* Using '/etc/pure-ftpd.conf' as main configuration file, since more
  option are available here.
* Adding pure-certd init script and config script for using SNI via FTP
* Splitting Uploadscript into it's own init script
* Simplified init script by removing much old stuff, which it seems,
  that it's not needed any more today. At least in my tests.
* Adding SVCNAME support.
* Removed sed for MAX_USER_LENGTH, as it's already default
* Removed DOCS for default docs
* Removed keepdir, since it looks like it's not needed anymore
* Overhauled eBuild

Signed-off-by: Conrad Kostecki <conikost@gentoo.org>

@Polynomial-C Could you please review and say your opinion on that changes? My main motivation was to bring init scripts for ```pure-certd```, as ```pure-certd``` only works via dedicated configuration file..

### CODE REVIEW

**Comment by Polynomial-C** [comment]
Looking at this PR as soon as I have recovered from a cold I currently have...

**Comment by ConiKost** [comment]
> Looking at this PR as soon as I have recovered from a cold I currently have...

I wish you a speedy recovery :-)

File: net-ftp/pure-ftpd/files/pure-uploadscript.initd [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/sbin/openrc-run
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+ftpd_configfile="/etc/${SVCNAME/uploadscript/ftpd}.conf"
+pidfile="/run/pure-ftpd.pid"
```
Comment (Polynomial-C) [inline net-ftp/pure-ftpd/files/pure-uploadscript.initd:6]: Should this init script really use the same pidfile like the `pure-ftpd.initd-r12` does?

File: net-ftp/pure-ftpd/pure-ftpd-1.0.50-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	newconfd "${FILESDIR}/pure-ftpd.confd-r4" pure-ftpd
+
+	newinitd "${FILESDIR}/pure-uploadscript.initd" pure-uploadscript
+	newconfd "${FILESDIR}/pure-uploadscript.confd" pure-uploadscript
+
+	if use implicittls ; then
+		sed -e '/^# Bind/s@21@990@' -i "${ED}"/etc/pure-ftpd.conf || die
+	fi
+
+	if use ssl; then
```
Comment (Polynomial-C) [inline net-ftp/pure-ftpd/pure-ftpd-1.0.50-r1.ebuild:121]: Please separate the semicolon with a space character from the USE flag like in the other `if` clauses.

File: net-ftp/pure-ftpd/pure-ftpd-1.0.50-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+		elog
+		elog "Before starting Pure-FTPd, you have to edit the /etc/pure-ftpd.conf file!"
+		elog
+		ewarn "It's *really* important to read the README provided with Pure-FTPd!"
+		ewarn "Check out http://download.pureftpd.org/pub/pure-ftpd/doc/README for general info"
+		ewarn "and http://download.pureftpd.org/pub/pure-ftpd/doc/README.TLS for SSL/TLS info."
+		ewarn
+	else
+		for v in ${REPLACING_VERSIONS}; do
+			if ver_test "${v}" -le "1.0.50"; then
```
Comment (Polynomial-C) [inline net-ftp/pure-ftpd/pure-ftpd-1.0.50-r1.ebuild:154]: Likewise (space in front of semicolon)

**Review by Polynomial-C — CHANGES_REQUESTED** [review]

File: net-ftp/pure-ftpd/pure-ftpd-1.0.50-r1.ebuild (hunk 3 above)
Comment (ConiKost) [reply net-ftp/pure-ftpd/pure-ftpd-1.0.50-r1.ebuild]: Done.

File: net-ftp/pure-ftpd/pure-ftpd-1.0.50-r1.ebuild (hunk 2 above)
Comment (ConiKost) [reply net-ftp/pure-ftpd/pure-ftpd-1.0.50-r1.ebuild]: Done.

File: net-ftp/pure-ftpd/files/pure-uploadscript.initd (hunk 1 above)
Comment (ConiKost) [reply net-ftp/pure-ftpd/files/pure-uploadscript.initd]: Indeed, a copy/paste error. Fixed now. Thanks!

**Comment by ConiKost** [comment]
Resovled merge conflict.

**Comment by Polynomial-C** [comment]
Sorry for not merging it earlier.

**Comment by ConiKost** [comment]
Great, many thanks :-)

**Pull request closed without merging**